### PR TITLE
[DVCSMP-2740] Added lock codes functionality to generic Z-Wave and Zi…

### PR DESCRIPTION
**Pull request by vaish-rahul**
…gbee Locks DTH

Added manufacturer specific DTHs for Yale Z-Wave and Zigbee Locks with
Lock Codes functionality and Advanced Settings features.

### CODE REVIEW

**Comment by vaish-rahul** [comment]
@jackchi Raised for PR for new DTHs for locks. Kindly approve.

**Comment by jackchi** [comment]
👍  Good to go for SmartLock Beta. QA for regression on ST & SC.

File: devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy [hunk 1]
```diff
@@ -85,90 +79,234 @@ metadata {
...
 }
 
+/**
+ * Returns the list of commands to be executed when the device is being configured/paired
+ *
+ */
+def doConfigure() {
+	log.trace "[DTH] Executing 'doConfigure()' for device ${device.displayName}"
+	state.configured = true
+	if (device.rawDescription.contains("mfr:003B")) {
```
Comment (mckeed) [inline devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy:147]: A better way to do this is `zwaveInfo.mfr == "003B"`

Comment (vaish-rahul) [reply devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy:147]: I'll make this change in the next commit. We are going to submit more changes to the DTH.

**Pull request merged**